src/str.rs

```diff
@@ -70,6 +89,16 @@ pub trait ParallelString {
     ///
     /// Note: the `Pattern` trait is private, for use only by Rayon itself.
     /// It is implemented for `char` and any `F: Fn(char) -> bool + Sync + Send`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let parts: Vec<_> = "((1 + 3) * 2)"
+    ///     .par_split_terminator(|c| c == '(' || c == ')')
+    ///     .collect();
+    /// assert_eq!(vec!["", "", "1 + 3", " * 2"], parts);
```

Comment:
I have to say, this is a peculiar example!  I do like that it shows a somewhat complicated pattern though.

Reply:
Yeah, I think by then I was kind of fed up of splitting simpler things! I also wanted to show how it would handle the trailing bracket differently to the leading ones.